src/Event/Http/HttpRequestEvent.php

```diff
@@ -20,15 +20,24 @@ final class HttpRequestEvent implements LambdaEvent
     private $headers;
     /** @var string */
     private $queryString;
+    /** @var float */
+    private $payloadVersion;
 
     public function __construct(array $event)
     {
-        if (! is_array($event) || ! isset($event['httpMethod'])) {
+        $this->payloadVersion = isset($event['version']) ? (float) $event['version'] : null;
```

Comment:
Ideally `$this->payloadVersion` should not be nullable. That would simplify code below.

Isn't the `version` always present?

One thing to consider: at this point, we are not sure if `$event` is a HTTP event`. What we could do is:

- first, check that it's a valid HTTP event (that's the lines below) -> if not, an exception is thrown
- then, we are sure it's a HTTP event, so we are sure we have `version` -> we can do `$this->payloadVersion = (float) $event['version']`

Reply:
I'll relocate the seting-part of the code to after the check RE ALB/API events so in theory it should always exist.